wgpu/src/backend/direct.rs

```diff
@@ -67,6 +75,16 @@ impl Context {
         self.0.create_adapter_from_hal(hal_adapter, PhantomData)
     }
 
+    #[cfg(any(not(target_arch = "wasm32"), feature = "emscripten"))]
```

Comment:
This isn't your fault, but this cfg block isn't right as it will fail on normal webgl2. It really should be `any(not(target_arch = "wasm32"), feature = "webgl2")`